src/coreclr/jit/gentree.cpp

```diff
@@ -5331,18 +5333,8 @@ bool GenTree::OperMayThrow(Compiler* comp)
 
         case GT_INTRINSIC:
             // If this is an intrinsic that represents the object.GetType(), it can throw an NullReferenceException.
-            // Report it as may throw.
-            // Note: Some of the rest of the existing intrinsics could potentially throw an exception (for example
-            //       the array and string element access ones). They are handled differently than the GetType intrinsic
-            //       and are not marked with GTF_EXCEPT. If these are revisited at some point to be marked as
-            //       GTF_EXCEPT,
-            //       the code below might need to be specialized to handle them properly.
-            if ((this->gtFlags & GTF_EXCEPT) != 0)
-            {
-                return true;
-            }
-
-            break;
+            // Currently, this is the only intrinsic that can throw an exception.
```

Comment:
The predicate that `OperMayThrow` implements is "does this node induce a new exception by itself". The `GTF_EXCEPT` check that was here before was a workaround for the lack of precise handling of intrinsics - it would return `true` for intrinsics (such as all the math ones) that themselves don't throw, but happened to have children that are throwing.

Additionally, bringing the check back would trigger the new assert in `fgValueNumberAddExceptionSet` that only treats the `GetType` intrinsic as possibly throwing.